allennlp/semparse/domain_languages/wikitables_language.py

```diff
@@ -462,123 +541,232 @@ def argmin(self, rows: List[Row], column: ComparableColumn) -> List[Row]:
+        if not all([isinstance(value, Date) for value in cell_values]):
+            raise ExecutionError(f"Invalid values for date selection function: {cell_values}")
+        return min(cell_values)  # type: ignore
 
     # These are some more number-column-specific functions, which only get added if we see a number
     # column.
 
-    def max(self, rows: List[Row], column: NumberColumn) -> Number:
+    def max_number(self, rows: List[Row], column: NumberColumn) -> Number:
         """
         Takes a list of rows and a column and returns the max of the values under that column in
         those rows.
         """
-        cell_row_pairs = self._get_number_row_pairs_to_filter(rows, column.name)
-        if not cell_row_pairs:
+        cell_values = [row.values[column.name] for row in rows]
+        if not cell_values:
             return 0.0  # type: ignore
-        return max([value for value, _ in cell_row_pairs])  # type: ignore
+        if not all([isinstance(value, Number) for value in cell_values]):
+            raise ExecutionError(f"Invalid values for number selection function: {cell_values}")
+        return max(cell_values)  # type: ignore
 
-    def min(self, rows: List[Row], column: NumberColumn) -> Number:
+    def min_number(self, rows: List[Row], column: NumberColumn) -> Number:
         """
         Takes a list of rows and a column and returns the min of the values under that column in
         those rows.
         """
-        cell_row_pairs = self._get_number_row_pairs_to_filter(rows, column.name)
-        if not cell_row_pairs:
+        cell_values = [row.values[column.name] for row in rows]
+        if not cell_values:
             return 0.0  # type: ignore
-        return min([value for value, _ in cell_row_pairs])  # type: ignore
+        if not all([isinstance(value, Number) for value in cell_values]):
+            raise ExecutionError(f"Invalid values for number selection function: {cell_values}")
+        return min(cell_values)  # type: ignore
 
     def sum(self, rows: List[Row], column: NumberColumn) -> Number:
         """
         Takes a list of rows and a column and returns the sum of the values under that column in
         those rows.
         """
-        cell_row_pairs = self._get_number_row_pairs_to_filter(rows, column.name)
-        if not cell_row_pairs:
+        cell_values = [row.values[column.name] for row in rows]
+        if not cell_values:
             return 0.0  # type: ignore
-        return sum([value for value, _ in cell_row_pairs])  # type: ignore
+        return sum(cell_values)  # type: ignore
 
     def average(self, rows: List[Row], column: NumberColumn) -> Number:
         """
         Takes a list of rows and a column and returns the mean of the values under that column in
         those rows.
         """
-        cell_row_pairs = self._get_number_row_pairs_to_filter(rows, column.name)
-        if not cell_row_pairs:
+        cell_values = [row.values[column.name] for row in rows]
```

Comment:
I do very much like this simplification, where the data has already been formatted to be the right type, so you don't have to check and reformat it all the time.  Should make this faster, too.